Sources/NIOHTTP1/HTTPDecoder.swift

```diff
@@ -234,6 +237,15 @@ public class HTTPDecoder<HTTPMessageT>: ByteToMessageDecoder, AnyHTTPDecoder {
         return response
     }
 
+    public func handlerRemoved(ctx: ChannelHandlerContext) {
+        if let buffer = self.cumulationBuffer?.slice(), buffer.readableBytes > 0, self.parser.upgrade == 1, ctx.channel.isActive {
+            self.cumulationBuffer = nil
+            ctx.fireChannelRead(NIOAny(buffer))
```

Comment:
is this really safe? It really feels like there's a scenario where the next channel hander expects HTTP messages and this gets invoked during a channel close and then the next handler crashes

Reply:
Yeah, don't do this if `ctx.channel.isActive` is `false`. That way if `handlerRemoved` is being invoked while the channel is being torn down you won't forward the data on.

Reply:
I legit didn't even see that. That's part of why I wanted this conditional factored out. ;)